src/formatDate.js

```diff
@@ -50,7 +50,49 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const currentSeparator = fromFormat[3];
+  const desiredSeparator = toFormat[3];
+  const currentFormat = fromFormat.slice(0, 3);
+  const desiredFormat = toFormat.slice(0, 3);
+  const dateArray = date.split(currentSeparator);
+  const day = dateArray[currentFormat.indexOf('DD')];
```

Comment:
Instead of manually accessing the date parts by their index, consider using array destructuring for more concise code.